test/event_hub.test.ts

```diff
@@ -239,6 +241,11 @@ describe("EventHub", () => {
     };
     expect(EventData).toEqual(expectedEvent);
   });
+
+  test("Disconnecting should disconnect the socket", () => {
+    eventHub.disconnect();
+    expect(disconnectCalled).toBe(true);
```

Comment:
```suggestion
    expect(eventHub._socketIo.discunnect).toBeCalled();
```

think you can do something like this, a bit more idiomatic

Reply:
No, I'm setting `_socketIo` to undefined as part of the disconnect.

Reply:
ah, I guess you then could do this:

```
const disconnectFn = vi.fn()
...
disconnect: disconnectFn,

expect(disconnectFn).toHaveBeenCalled()
```

but we can go with this approach as well